tools/cbmc/include/cbmc.h

```diff
@@ -26,12 +28,25 @@ enum CBMC_LOOP_CONDITION { CBMC_LOOP_BREAK, CBMC_LOOP_CONTINUE, CBMC_LOOP_RETURN
 #define __CPROVER_printf_ptr(var) { uint8_t *ValueOf_ ## var = (uint8_t *) var; }
 #define __CPROVER_printf2_ptr(str,exp) { uint8_t *ValueOf_ ## str = (uint8_t *) (exp); }
 
-/* xWantedSize is not bounded in this function, but there might be a need to bound it in the future.
- * In theory, CBMC malloc allows to allocate an arbitrary amount of data. This will not be true for
+/* CBMC assert to test pvPortMalloc result when xWantedSize is
+ * 0. Mostly used to report full coverage on pvPortMalloc, but use
+ * with caution as it might complicate debugging
+ */
+#define __CPROVER_assert_zero_allocation() \
+        __CPROVER_assert( pvPortMalloc(0) == NULL, \
+			  "pvPortMalloc allows zero-allocated memory.")
+
+/* xWantedSize is not bounded in this function, but there might be a
+ * need to bound it in the future.  In theory, CBMC malloc allows to
+ * allocate an arbitrary amount of data. This will not be true for
```

Comment:
Should an upper bound be expressed in this situation then?

Reply:
I'm updating the comment to read:

```
/* 
 * A stub for pvPortMalloc that nondeterministically chooses to return
 * either NULL or an allocation of the requested space.  The stub is
 * guaranteed to return NULL when asked to allocate 0 bytes.  
 * This stub is used in some of the TaskPool proofs.
 */
```